### PR TITLE
feat: extend the creator using the site-utilities Editor abstract class

### LINKED ISSUE #3564: Add Monaco editor to the Creator and Explore sites
# Abstract

The `Creator` and `Component explorer` were created with the idea in mind that they would be useful to both developers and designers as a creation and preview sites for the components from the FAST project. As such, it currently has a suite of UI that allows for the navigation of data, editing of data and previewing of data. However there is a lack of code preview that should be present for developers to be able to copy code to use it in their applications.

## Requirements

Resolve the following issues:
- Convert the data (dictionary data from the Message System) into a format that [Monaco](https://microsoft.github.io/monaco-editor/index.html) can interpret and show as code #3561
- Convert any editing done to the code back into data (dictionary data from the Message System) #3561
- Add intellisense #3563

### PR DESCRIPTION
**Pull request by janechu**
# Description

This change converts the `<Creator />` component of the creator app to use the abstracted `<Editor />` from the `@microsoft/site-utilities` private package.

This does break the contract for the `ProjectFile`, which is changing from using the `views` to a single `DataDictionary`.

This change enables:
- Updated style per the Explore site
- Mobile views

## Motivation & context

Closes #3564

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: sites/fast-creator/app/creator.tsx [hunk 1]
```diff
@@ -111,254 +87,199 @@ class Creator extends Foundation<CreatorHandledProps, {}, CreatorState> {
...
-                </Container>
+                        </Background>
+                    </div>
+                </div>
+                <div className={this.paneEndClassNames}>
+                    <ModularForm
+                        messageSystem={this.fastMessageSystem}
+                        controls={[
+                            new StandardControlPlugin({
+                                type: ControlType.linkedData,
```
Comment (nicholasrice) [inline sites/fast-creator/app/creator.tsx:265]: Should this instance be cached so you don't have to new up a `StandardControlPlugin` every render?

File: sites/fast-creator/app/preview.tsx [hunk 2]
```diff
@@ -231,6 +233,7 @@ class Preview extends Foundation<{}, {}, PreviewState> {
                         );
                         break;
                     case MessageSystemType.custom:
+                        console.log("ugh", messageData);
```
Comment (nicholasrice) [inline sites/fast-creator/app/preview.tsx:236]: *sigh*

**Review by nicholasrice — APPROVED** [review]
Looks good aside from above comments.

File: sites/fast-creator/app/preview.tsx (hunk 2 above)
Comment (janechu) [reply sites/fast-creator/app/preview.tsx:236]: Just, uh, checking to see if reviewers were paying attention...

Comment (janechu) [inline sites/fast-creator/app/preview.tsx:236]:
```suggestion
```

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**